RetailCoder.VBE/UI/Command/FindAllReferencesCommand.cs

```diff
@@ -160,9 +162,53 @@ private Declaration FindTarget(object parameter)
                 return declaration;
             }
 
+            return _vbe.ActiveCodePane != null && (_vbe.SelectedVBComponent?.HasDesigner ?? false)
+                    ? FindFormDesignerTarget()
+                    : FindCodePaneTarget();
+        }
+
+        private Declaration FindCodePaneTarget()
+        {
             return _state.FindSelectedDeclaration(_vbe.ActiveCodePane);
         }
 
+        private Declaration FindFormDesignerTarget(QualifiedModuleName? qualifiedModuleName = null)
+        {            
+            var projectId = qualifiedModuleName.HasValue 
+                                               ? qualifiedModuleName.Value.ProjectId 
+                                               :  _vbe.ActiveVBProject.ProjectId;
+
+            var component = qualifiedModuleName.HasValue
+                                               ? qualifiedModuleName.Value.Component
+                                               :_vbe.SelectedVBComponent;
+                        
+
+            if (component?.HasDesigner ?? false)
+            {
+                if (qualifiedModuleName.HasValue)
+                {
+                    return _state.DeclarationFinder.MatchName(qualifiedModuleName.Value.Name)
+                                                   .SingleOrDefault(m => m.ProjectId == projectId
+                                                                      && m.DeclarationType.HasFlag(qualifiedModuleName.Value.ComponentType)
+                                                                      && m.ComponentName == component.Name);
+                }
+
+                var selectedCount = component.SelectedControls.Count;                
+                if (selectedCount > 1) { return null; }
+
+                // Cannot use DeclarationType.UserForm, parser only assigns UserForms the ClassModule flag
```

Comment:
:+1:

Reply:
I don't follow, sorry... I'm filtering by component declaration type, which would ideally be UserForm in the case of forms, but the parser doesn't flag them as anything more specific than ClassModule.

What am I missing?